QModInstaller/QModInjector.cs

```diff
@@ -32,117 +30,95 @@ public QModInjector(string dir, string managedDir = null)
             {
-                // if a backup exists
                 if (File.Exists(backupFilename))
                 {
-                    // remove the dirty dll
                     File.Delete(mainFilename);
 
-                    // move the backup into its place
                     File.Move(backupFilename, mainFilename);
 
-                    return true;
+                    Console.WriteLine();
+                    Console.WriteLine("QModManager uninstalled successfully");
+                    Console.WriteLine();
+                    Console.WriteLine("Press any key to exit...");
+                    Console.ReadKey();
+                    Environment.Exit(0);
                 }
 
-                return false;
-            }
-            catch (IOException e)
-            {
-                Console.WriteLine(e.Message);
-                Console.WriteLine(e.StackTrace);
-                if (e.InnerException != null)
-                {
-                    Console.WriteLine(e.InnerException.Message);
-                    Console.WriteLine(e.InnerException.StackTrace);
-                }
+                Console.WriteLine();
+                Console.WriteLine("Cannot uninstall, file 'Assembly-CSharp-qoriginal.dll' doesn't exist");
+                Console.WriteLine("To uninstall, you will need to verify game contents in steam");
+                Console.WriteLine();
+                Console.WriteLine("Press any key to exit...");
                 Console.ReadKey();
                 Environment.Exit(0);
-                return false;
+            }
+            catch (Exception e)
+            {
+                ExceptionUtils.ParseException(e);
             }
         }
 
-
-        private bool isInjected()
+        public bool IsInjected()
         {
             try
             {
                 var game = AssemblyDefinition.ReadAssembly(mainFilename);
 
-                var rClass = "TankCamera";
-                var rMethod = "Awake";
-
-                TypeDefinition type = null;
-                MethodDefinition method = null;
-
-                type = game.MainModule.GetType(rClass);
-                method = type.Methods.First(x => x.Name == rMethod);
-
-                var installer = AssemblyDefinition.ReadAssembly(installerFilename);
-                var patchMethod = installer.MainModule.GetType("QModInstaller.QModPatcher").Methods.FirstOrDefault(x => x.Name == "Patch");
+                AssemblyDefinition installer = AssemblyDefinition.ReadAssembly(installerFilename);
+                MethodDefinition patchMethod = installer.MainModule.GetType("QModInstaller.QModPatcher").Methods.Single(x => x.Name == "Patch");
 
-                bool patched = false;
+                TypeDefinition type = game.MainModule.GetType("TankCamera");
+                MethodDefinition method = type.Methods.Single(x => x.Name == "Awake");
 
                 foreach (var instruction in method.Body.Instructions)
                 {
```

Comment:
Double check the namespace/class name of the method call in the IL instruction check below

Reply:
At first, I was thinking about removing the old patch (if there was one), then adding the new patch call. I changed my mind and I added the old namespace back with a dummy method which calls the original one. I will move it to another file

Reply:
If that's not what you are talking about, please clarify

Reply:
It's mainly just to make sure the correct method is there after all these other changes.  
But if you have the opportunity to simplify things even further, by not needing to hold onto the old method at all, I would try it out and test it.